c/core.go

```diff
@@ -79,7 +80,15 @@ func NewWithNotifyStart(
 	startFn func(context.Context, NotifyStartFn) error,
 	opts ...Opt,
 ) ChildSpec {
-	spec := ChildSpec{}
+	spec := ChildSpec{
+		// Child workers by default will have 5 seconds to terminate before
+		// reporting a timeout error as specified on the Erlang OTP documentation.
+		// http://erlang.org/doc/design_principles/sup_princ.html#tuning-the-intensity-and-period
+		//
+		// Note there is no brutal kill from the supervisor as Go (for better or
+		// worse), does not offer guaranteed kill signals for goroutines.
```

Comment:
So, golang doesn't offer you a way to _*kill*_ a goroutine, you can stop it via `context`, as long as the goroutine is doing a  `<-context.Done()`, which, is not guaranteed. In the scenarios where the goroutine does not stop after the timeout (`terminateCh` doesn't return), in an environment like Erlang or Haskell, you can hard kill the thread, in golang, you cannot. In our implementation, we call it done with a timeout error and move forward to the next thread to stop. Not ideal, but that's the only way it can be.